### PR TITLE
Add a LFS nimble task and gitignore of LFS and state sim

### PR DESCRIPTION
**Pull request by mratsim**
ignore new folders introduced in #339 and the json files produced by the state sim

### CODE REVIEW

File: beacon_chain.nimble [hunk 1]
```diff
@@ -54,3 +54,7 @@ task test, "Run all tests":
 
   # State sim; getting into 3rd epoch useful
   buildBinary "state_sim", "research/", "-r -d:release", "--validators=128 --slots=140"
+
+task sync_lfs_tests, "Sync LFS json tests":
+  # Syncs the json test files (but not the EF yaml tests)
+  exec "scripts/process_lfs.sh jsonTestsCache"
```
Comment (stefantalpalaru) [inline beacon_chain.nimble]: You don't need the cache dir outside CI VMs. Simply running `scripts/process_lfs.sh` repeatedly on the same repo will only download LFS files the first time and whenever the corresponding submodule is bumped.

Now this script is called by the `test` target in the Makefile. Why do you also need it in a *.nimble task?

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,9 @@ build/
 
 /scripts/testnet*.sh
 
+# LFS cache
+jsonTestsCache/
+tmpcache/
```
Comment (stefantalpalaru) [inline .gitignore]: This is deleted by the script.

File: .gitignore [hunk 3]
```diff
@@ -12,3 +12,9 @@ build/
...
+
+# State sim # TODO - move in another folder
+0000-*.json
```
Comment (stefantalpalaru) [inline .gitignore]: Yes, we can ignore them, but they're still there, adding noise to directory listings. Better to delete them on successful test runs.

File: beacon_chain.nimble (hunk 1 above)
Comment (mratsim) [reply beacon_chain.nimble]: this is for local development and also linked to limiting our reliance on shell scripts #343

File: .gitignore (hunk 2 above)
Comment (mratsim) [reply .gitignore]: OK

File: beacon_chain.nimble (hunk 1 above)
Comment (stefantalpalaru) [reply beacon_chain.nimble]: > this is for local development

So is `make test`.

**Comment by mratsim** [comment]
Merging for now, we can remove them when we change the repo structure to not rely on a parent nimbus config.

**Pull request merged**